source/mongodb-handshake/handshake.rst

```diff
@@ -224,6 +224,49 @@ Example::
         - "Oracle JVM EE 9.1.1"
 
 
+Supporting Wrapping Libraries
+=============================
+
+Drivers MUST allow libraries which wrap the driver to append to the client
+metadata generated by the driver. The following class definition defines the
+options which MUST be supported:
+
+.. code:: typescript
+
+  class DriverInfoOptions {
+    /**
+     * The name of the library wrapping the driver.
+     */
+    name: Optional<String>;
```

Comment:
This will address my comment in https://github.com/mongodb/specifications/pull/687#discussion_r345940014 about possible issues matching up the exploded, delimited values if one was omitted. I might also suggest requiring them to be non-empty strings and also not contain the delimiter character.

Reply:
I agree about name, but some wrappers may not be able to report their version.  I don't think Morphia, for example, has a concept of its own version that is accessible at runtime.

Reply:
Made `name` non-optional